src/components/Form/Form.js

```diff
@@ -13,16 +13,49 @@ const form = () => {
   const { quizForm, step } = formState;
 
   const nextStep = () => {
+    const currStep = step + 1;
     updateFormState({
       ...formState,
-      step: step + 1
+      step: currStep
     });
+
+    initDefaultCheckedInput(quizForm[formState.step]['values'], step, currStep);
   };
 
   const prevStep = () => {
+    const currStep = step - 1;
     updateFormState({
       ...formState,
-      step: step - 1
+      step: currStep
+    });
+
+    initDefaultCheckedInput(quizForm[formState.step]['values'], step, currStep);
+  };
+
+  const initDefaultCheckedInput = (currentFormPage, step, currStep) => {
+    let updateFormPage = { ...currentFormPage };
+    const updatedForm = {
+      ...formState
+    };
+
+    for (let key in updateFormPage) {
+      if ('text' !== updateFormPage[key]['elementConfig']['type']) {
```

Comment:
this looks structurally intimidating at first glance ... 
https://github.com/crisanlucid/react-quiz/blob/c3ca710db370d6c57130a2a964b72ac2f0dc460d/src/components/Form/Form.js#L42-L51

can you try to write the code that the reader is able to understand what it does without parsing each line ?